backends/postgres.go

```diff
@@ -44,6 +51,51 @@ func (p *PostgresBackend) CreateToken(token string) error {
 	return err
 }
 
+func (p *PostgresBackend) IncrementAndGetNamespacedToken(token string, namespace string) (int64, error) {
+	tx, err := p.db.Begin()
+	if err != nil {
+		defer tx.Rollback()
+		log.Printf("Error starting transaction: %s", err)
+		return 0, errDatabase
+	}
+
+	var count int64
+
+	err = tx.QueryRow(`SELECT s.count
+						FROM namespaced_counters AS s
+						WHERE token = $1 AND namespace = $2 FOR UPDATE`, token, namespace).Scan(&count)
+
+	// No counter found for this namespace, create namespace and initialize count to 0
+	if err == sql.ErrNoRows {
+		_, err := p.db.Exec(`INSERT INTO namespaced_counters (token, namespace, count)
+							 VALUES ($1, $2, 0)`, token, namespace)
+		if err != nil {
+			defer tx.Rollback()
+			log.Printf("Error creating namespaced counter: %s", err)
+			return 0, errDatabase
+		}
+
+		tx.Commit()
+
+		return 0, nil
+	} else {
```

Comment:
You don't need this else since you're always returning from the other part of the `if`